src/main/scala/com/cibo/scalastan/StanType.scala

```diff
@@ -376,11 +404,17 @@ trait StanVectorLike extends StanVectorOrMatrix {
   private[scalastan] def getData(data: Seq[Double]): Seq[String] = data.map(_.toString)
   private[scalastan] def getDims(data: Seq[Double]): Seq[Int] = Seq(data.length)
 
-  private[scalastan] def parse(name: String, values: Map[String, String]): Seq[Double] = {
+  private[scalastan] def parse(
+    name: String,
+    parameterChains: Map[String, Vector[Vector[String]]]
+  ): Vector[Vector[Vector[Double]]] = {
     val prefix = s"$name."
-    values.filterKeys(_.startsWith(prefix)).toVector.map { case (key, value) =>
-      key.slice(prefix.length, key.length).toInt -> value
-    }.sortBy(_._1).map(_._2.toDouble)
+    val sorted = parameterChains.keys.filter(_.startsWith(prefix)).toVector.sortBy(_.drop(prefix.length).toInt)
+    parameterChains.head._2.zipWithIndex.map { case (chain, chainIndex) =>
```

Comment:
Is there enough duplication in these parse methods at least for array and vector to warrant delegating to a common method?  Or some common utility methods?